sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ConnectionString/ConnectionStringParser.cs

```diff
@@ -0,0 +1,124 @@
+
+        /// <summary>
+        /// Will evaluate connection string and return the requested endpoint.
+        /// </summary>
+        /// <remarks>
+        /// Parsing the connection string MUST follow these rules:
+        ///     1. check for explicit endpoint (location is ignored)
+        ///     2. check for endpoint suffix (location is optional)
+        ///     3. use default endpoint (location is ignored)
+        /// This behavior is required by the Connection String Specification.
+        /// </remarks>
+        internal static string GetIngestionEndpoint(this Azure.Core.ConnectionString connectionString)
+        {
+            // Passing the user input values through the Uri constructor will verify that we've built a valid endpoint.
+            Uri uri;
+
+            if (connectionString.TryGetNonRequiredValue(Constants.IngestionExplicitEndpointKey, out string explicitEndpoint))
+            {
+                if (!Uri.TryCreate(explicitEndpoint, UriKind.Absolute, out uri))
+                {
+                    throw new ArgumentException($"The value for {Constants.IngestionExplicitEndpointKey} is invalid. '{explicitEndpoint}'");
+                }
+            }
+            else if (connectionString.TryGetNonRequiredValue(Constants.EndpointSuffixKey, out string endpointSuffix))
+            {
+                var location = connectionString.GetNonRequired(Constants.LocationKey);
+                if (!TryBuildUri(prefix: Constants.IngestionPrefix, suffix: endpointSuffix, location: location, uri: out uri))
+                {
+                    throw new ArgumentException($"The value for {Constants.EndpointSuffixKey} is invalid. '{endpointSuffix}'");
+                }
+            }
+            else
+            {
+                return Constants.DefaultIngestionEndpoint;
+            }
+
+            return uri.AbsoluteUri;
+        }
+
+        /// <summary>
+        /// Construct a Uri from the possible parts.
+        /// Format: "location.prefix.suffix".
+        /// Example: "https://westus2.dc.applicationinsights.azure.cn/".
+        /// </summary>
+        /// <remarks>
+        /// Will also attempt to sanitize user input. Won't fail if the user typo-ed an extra period.
+        /// </remarks>
+        internal static bool TryBuildUri(string prefix, string suffix, out Uri uri, string location = null)
+        {
+            // Location and Suffix are user input fields and need to be sanitized (extra spaces or periods).
+            char[] trimPeriod = new char[] { '.' };
+
+            if (location != null)
+            {
+                location = location.Trim().TrimEnd(trimPeriod);
+
+                // Location names are expected to match Azure region names. No special characters allowed.
+                if (!location.All(x => char.IsLetterOrDigit(x)))
+                {
+                    throw new ArgumentException($"The value for Location must not contain special characters. '{location}'");
```

Comment:
>special characters [](start = 90, length = 18)

`special characters` here might be ambiguous, say `must only contain degit or letter`?

Reply:
done